pkgs/os-specific/linux/kernel/update-zen.py

```diff
@@ -78,6 +78,26 @@ def update_file(relpath, variant, version, suffix, sha256):
             print(result, end='')
 
 
+def read_file(relpath, variant):
+    file_path = os.path.join(DIR, relpath)
+    re_version = re.compile(fr'^\s*version = "(.+)"; #{variant}')
+    re_suffix = re.compile(fr'^\s*suffix = "(.+)"; #{variant}')
+    version = None
+    suffix = None
+    with fileinput.FileInput(file_path, mode='r') as f:
+        for line in f:
+            version_match = re_version.match(line)
+            if version_match:
+                version = version_match.group(1)
+                continue
+
+            suffix_match = re_suffix.match(line)
+            if suffix_match:
+                suffix = suffix_match.group(1)
+                continue
```

Comment:
Maybe add something like this:
```python
    anychecked = False
    with fileinput.FileInput(file_path, mode='r') as f:
        for line in f:
            version_match = re_version.match(line)
            if version_match:
                version = version_match.group(1)
                if anychecked:
                	break
                else:
                	anychecked = True
	                continue

            suffix_match = re_suffix.match(line)
            if suffix_match:
                suffix = suffix_match.group(1)
                if anychecked:
                	break
                else:
                	anychecked = True
	                continue
```

Reply:
You mean to bail out early? Yeah, I didn't do it in the first place since the file is small enough, but I've added the following (imo, a good compromise between optimization and readability):

```
if version and suffix:
    break 
```